model/src/main/kotlin/PackageProvider.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
```

Comment:
Is this supposed to be used only from `ClearlyDefinedPackageCurationProvider`? If so, shouldn't it be put next to it (in the analyzer) ?

Reply:
Strictly speaking #155 is about adding the "(package) provider" concept to ORT independently of ClearlyDefined. So while I agree the primary use-case to be in context with ClearlyDefined, I do not want to rule out other use-cases at this point by moving the class.

Reply:
But you're right that I should separate the ClearlyDefined-specific code more from the enum. Will do.